src/control/server/storage/bdev/backend.go

```diff
@@ -216,21 +217,57 @@ func getController(pciAddr string, bcs []spdk.Controller) (*storage.NvmeControll
 	return scs[0], nil
 }
 
-func (b *spdkBackend) Format(pciAddr string) (*storage.NvmeController, error) {
-	if err := b.Init(); err != nil {
-		return nil, err
+func (b *spdkBackend) formatNvmeBdev(dev string, req FormatRequest, resp *DeviceFormatResponse) error {
+	spdkOpts := spdk.EnvOptions{
+		ShmID:        req.ShmID,
+		MemSize:      req.MemSize,
+		PciWhiteList: []string{dev},
+		EnableVMD:    b.IsVmdEnabled(),
 	}
 
-	ctrlr, err := getController(pciAddr, b.binding.controllers)
-	if err != nil {
-		return nil, err
+	b.log.Debugf("spdk init opts: %+v", spdkOpts)
+
+	// provide bdev as whitelist so only formatting device is bound
+	if err := b.binding.InitSPDKEnv(spdkOpts); err != nil {
+		return errors.Wrap(err, "failed to initialize SPDK")
```

Comment:
I don't feel strongly either way, but out of curiosity will this lock us into not being able to support parallel formatting of NVMe devices?

Reply:
no . I think it rather enables it, if we can bind only to one controller then we can spawn multiple goroutines to each do init+format on distinct controllers in parallel

Reply:
Hmm, but the binding is per-process, not per-goroutine, right? In any case, I'm not too worried about it now, was just wondering.

Reply:
so in the provider you would spawn one go-routine per controller and each of those would call backend binding to single controller (one daos_admin process per go-routine, each formatting single controller in parallel)